extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -286,11 +339,34 @@ export class Bind {
     // Helper function for scanning the tree walker's next node.
     // Returns true if the walker has no more nodes.
     const scanNextNode_ = () => {
-      const element = walker.nextNode();
-      if (!element) {
+      const node = walker.currentNode;
+      if (!node) {
         return true;
       }
+      // Walker is filtered to only return elements
+      const element = dev().assertElement(node);
       const tagName = element.tagName;
+      if (DYNAMIC_TAGS.includes(tagName)) {
+        // Listen for changes in amp-mustache templates
+        // Templated HTML is added as a sibling to the template tag.
+        // So observe the parent.
+        // TODO(kmh287): What if parent is the body tag?
+        // TODO(kmh287): Generify logic for node observation strategy
+        // when bind supprots more dynamic nodes.
+        if (tagName == 'TEMPLATE') {
+          const templateParent = element.parentElement;
+          if (templateParent) {
+            this.subtreeMutationObserver_.observe(templateParent, {
+              childList: true,
+            });
+            this.dynamicRoots_.push(templateParent);
+          }
+        } else {
```

Comment:
These two cases are essentially the same, save for the `root`. Mind cleaning this up?